applications/FluidDynamicsApplication/python_scripts/fluid_solver.py

```diff
@@ -259,3 +273,162 @@ def _SetNodalProperties(self):
         # Transfer the obtained properties to the nodes
         KratosMultiphysics.VariableUtils().SetVariable(KratosMultiphysics.DENSITY, rho, self.main_model_part.Nodes)
         KratosMultiphysics.VariableUtils().SetVariable(KratosMultiphysics.VISCOSITY, kin_viscosity, self.main_model_part.Nodes)
+
+    # TODO: I THINK THIS SHOULD BE MOVED TO THE BASE PYTHON SOLVER
+    def is_restarted(self):
+        # this function avoids the long call to ProcessInfo and is also safer
+        # in case the detection of a restart is changed later
+        return self.main_model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED]
+
+    def get_estimate_dt_utility(self):
+        if not hasattr(self, '_estimate_dt_utility'):
```

Comment:
```suggestion
        if not hasattr(self, '__estimate_dt_utility'):
```

this is private

Reply:
Wouldn't this give name mangling errors when calling the method from the derived solvers?

Reply:
this is the attribute and not the method
with the attribute it is INTENDED that you get the mangling problems, since you should use the get... function

Reply:
OFC. (stupid question)

Reply:
I've just tested and to use the `hasattr` we would need to do something like  `'class_name'+'_estimate_dt_utility'`.
Otherwise it enters the create methods each time.

Reply:
ok thx for checking